client/tests/routes/operations/Operations.test.tsx

```diff
@@ -1,64 +1,54 @@
 import Operations from "@/app/components/routes/operations/Operations";
 import "@testing-library/jest-dom";
 import { render, screen } from "@testing-library/react";
-import fetchMock from "jest-fetch-mock";
+import { actionHandler, useSession } from "@/tests/setup/mocks";
+import { Session } from "@/tests/setup/types";
+
+useSession.mockReturnValue({
+  data: {
+    user: {
+      app_role: "cas_admin",
+    },
+  },
+} as Session);
 
 describe("Operations component", () => {
-  beforeEach(() => {
-    jest.restoreAllMocks();
-    fetchMock.enableMocks(); // Enable fetch mocking
+  beforeEach(async () => {
+    vi.clearAllMocks();
   });
   it("renders the Operations grid", async () => {
-    fetchMock.mockResponse(
-      JSON.stringify([
+    actionHandler.mockReturnValueOnce({
+      data: [
         {
-          id: 1,
-          name: "Operation 1",
-          type: "Single Facility Operation",
-          naics_code: 1,
-          previous_year_attributable_emissions: "100",
-          swrs_facility_id: "1001",
-          bcghg_id: "123",
-          opt_in: false,
-          operator: 1,
+          id: "1",
+          bc_obps_regulated_operation: "1",
+          operator: "1",
+          submission_date: "2021-10-01T00:00:00",
           status: "Not Started",
-          reporting_activities: [],
-          regulated_products: [],
-          documents: [],
-          contacts: [],
-          operator_id: 1,
-          naics_code_id: 1,
+          name: "Operation 1",
+          bcghg_id: "1",
         },
         {
-          id: 2,
-          name: "Operation 2",
-          type: "Type B",
-          naics_code: 2,
-          permit_issuing_agency: "authority",
-          permit_number: "256",
-          previous_year_attributable_emissions: null,
-          swrs_facility_id: null,
-          bcghg_id: null,
-          opt_in: false,
-          operator: 2,
+          id: "2",
+          bc_obps_regulated_operation: "2",
+          operator: "2",
+          submission_date: "2021-10-01T00:00:00",
           status: "Not Started",
-          reporting_activities: [],
-          regulated_products: [],
-          documents: [],
-          contacts: [],
-          operator_id: 2,
-          naics_code_id: 2,
+          name: "Operation 2",
+          bcghg_id: "2",
         },
-      ]),
-    );
+      ],
+      row_count: 2,
+    });
     render(await Operations());
+
     // Check if the grid of mock data is present
-    await expect(screen.getByText(/Operation 1/i)).toBeVisible();
-    await expect(screen.getByText(/Operation 2/i)).toBeVisible();
+    expect(screen.getByText(/Operation 1/i)).toBeVisible();
+    expect(screen.getByText(/Operation 2/i)).toBeVisible();
     // temporarily commented out because render only renders half the grid
```

Comment:
Do we still need this comment? I think the whole grid must be rendering now if we can assert the start registration link is visible